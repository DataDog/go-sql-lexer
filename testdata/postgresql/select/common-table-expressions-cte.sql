WITH recursive_subordinates AS (
    SELECT id, manager_id FROM employees WHERE id = 1
    UNION ALL
    SELECT e.id, e.manager_id FROM employees e INNER JOIN recursive_subordinates rs ON rs.id = e.manager_id
)
SELECT * FROM recursive_subordinates;
