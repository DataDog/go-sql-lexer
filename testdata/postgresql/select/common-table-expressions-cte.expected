[
    {
        "expected": "WITH recursive_subordinates AS ( SELECT id, manager_id FROM employees WHERE id = ? UNION ALL SELECT e.id, e.manager_id FROM employees e INNER JOIN recursive_subordinates rs ON rs.id = e.manager_id ) SELECT * FROM recursive_subordinates",
        "statement_metadata": {
            "size": 41,
            "tables": [
                "employees",
                "recursive_subordinates"
            ],
            "commands": [
                "SELECT",
                "JOIN"
            ],
            "comments": [],
            "procedures": []
        }
    }
]